Testing/SystemTests/tests/analysis/GSASIIRefineFitPeaksTest.py

```diff
@@ -0,0 +1,168 @@
+                try:
+                    path = _GSASFinder._find_directory_by_name(cur_dir_name=child, cur_dir_path=child_path,
+                                                               level=level + 1, name_to_find=name_to_find,
+                                                               max_level=max_level)
+                except OSError:  # Probably "Permission denied". Either way, just ignore it
+                    pass
+                else:
+                    if path is not None:
+                        return path
+
+    @staticmethod
+    def _path_to_g2conda():
+        """
+        Find the g2conda directory (where GSAS-II normally sits), as long as it exists less than 3 levels away from
+        the root directory
+        """
+        root_directory = os.path.abspath(os.sep)
+        return _GSASFinder._find_directory_by_name(cur_dir_path=root_directory, cur_dir_name=root_directory, level=0,
+                                                   name_to_find="g2conda", max_level=3)
+
+    @staticmethod
+    def GSASIIscriptable_location():
+        """
+        Find the path to GSASIIscriptable.py, if it exists and is less than 2 levels away from the root directory
+        """
+        path_to_g2conda = _GSASFinder._path_to_g2conda()
+        print(path_to_g2conda)
+        if path_to_g2conda is None:
+            return None
+
+        if os.path.isfile(os.path.join(path_to_g2conda, "GSASIIscriptable.py")):
+            return path_to_g2conda
+
+        if os.path.isfile(os.path.join(path_to_g2conda, "GSASII", "GSASIIscriptable.py")):
+            return os.path.join(path_to_g2conda, "GSASII")
+
+        return None
+
+
+class _GSASIIRefineFitPeaksTestHelper(object):
+
+    _LATTICE_PARAM_TBL_NAME = "LatticeParameters"
+    _INPUT_WORKSPACE_FILENAME = "focused_bank1_ENGINX00256663.nxs"
+    _PHASE_FILENAME = "FE_ALPHA.cif"
+    _INST_PARAM_FILENAME = "template_ENGINX_241391_236516_North_bank.prm"
+    _TEMP_DIR = tempfile.gettempdir()
+
+    def path_to_gsas(self):
+        return _GSASFinder.GSASIIscriptable_location()
+
+    def input_ws_path(self):
+        return mantid.FileFinder.getFullPath(self._INPUT_WORKSPACE_FILENAME)
+
+    def phase_file_path(self):
+        return mantid.FileFinder.getFullPath(self._PHASE_FILENAME)
+
+    def inst_param_file_path(self):
+        return mantid.FileFinder.getFullPath(self._INST_PARAM_FILENAME)
+
+    def remove_all_gsas_files(self, gsas_filename_without_extension):
+        for filename in os.listdir(self._TEMP_DIR):
+            if re.search(gsas_filename_without_extension, filename):
+                os.remove(os.path.join(self._TEMP_DIR, filename))
+
+
+class GSASIIRefineFitPeaksRietveldTest(stresstesting.MantidStressTest, _GSASIIRefineFitPeaksTestHelper):
+
+    rwp = None
+    gof = None
+    gsas_proj_path = None
+    _REFERENCE_FILE_NAME = "GSASIIRefineFitPeaksRietveldFitParams.nxs"
+    _GSAS_PROJ_FILE_NAME = "GSASIIRefineFitPeaksRietveldTest.gpx"
+
```

Comment:
I think we also need to add a `skipTests` method, see [here](https://github.com/mantidproject/mantid/blob/0a677423d0f19c332481ba069fa60b19a7ca0dd4/Testing/SystemTests/tests/analysis/ISIS_PowderPearlTest.py#L53) for an example, that skips the tests if GSAS is not available. Most developers won't have this installed and all of the standard nightly jobs won't have it. Essentially we can probably just pull up the `path_to_gsas` bit below into `skipTests`.